src/network/teldat/snmp/mode/cellsradio.pm

```diff
@@ -0,0 +1,310 @@
+    rsrp        => { oid => '.1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.22' }, # teldatCellularStateMobileRxRSRP : Cellular mobile reference symbol received power (RSRP).
+    rsrq        => { oid => '.1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.23' }, # teldatCellularStateMobileRxRSRQ : Cellular mobile reference signal received quality (RSRQ).
+    snr         => { oid => '.1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.24' }, # teldatCellularStateMobileRxSINR : Cellular mobile signal versus noise ratio (SINR).
+    simStatus   => { oid => '.1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1.26' } # teldatCellularStateMobileSIMStatus : Cellular mobile SIM status.
+};
+
+my $oid_teldatCellularInfoInterfaceEntry = '.1.3.6.1.4.1.2007.4.1.2.2.2.18.1.1'; # teldatInfoInterfaceTable
+my $oid_teldatCellularStateMobileEntry = '.1.3.6.1.4.1.2007.4.1.2.2.2.18.3.2.1'; # teldatStateMobileTable
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    $self->{global} = { detected => 0 };
+    $self->{cells} = {};
+
+    my $snmp_result = $options{snmp}->get_multiple_table(
+        oids => [
+            { oid => $mapping_state_interface->{interfaceState}->{oid} },
+            { oid => $oid_teldatCellularInfoInterfaceEntry, start => $mapping_info_interface->{imei}->{oid}, end => $mapping_info_interface->{simId}->{oid} },
+            { oid => $oid_teldatCellularStateMobileEntry, start => $mapping_state_mobile->{techno}->{oid}, end => $mapping_state_mobile->{simStatus}->{oid} }
+        ],
+        nothing_quit => 1
+    );
+
+    foreach my $oid (keys %{$snmp_result->{$oid_teldatCellularInfoInterfaceEntry}}) {
+        next if ($oid !~ /^$mapping_info_interface->{imei}->{oid}\.(.*)$/);
+        my $instance = $1;
+
+        my $result = $options{snmp}->map_instance(mapping => $mapping_info_interface, results => $snmp_result->{$oid_teldatCellularInfoInterfaceEntry}, instance => $instance);
+        my $result2 = $options{snmp}->map_instance(mapping => $mapping_state_interface, results => $snmp_result->{$mapping_state_interface->{interfaceState}->{oid}}, instance => $instance);
+
+        my $cell_id = $result->{imei} =~ /^(?:[0-9]+)$/ ? $result->{imei} : $result->{simId};
+        next if ($cell_id !~ /^(?:[0-9]+)$/);
+        next if (defined($self->{option_results}->{filter_cell_id}) && $self->{option_results}->{filter_cell_id} ne '' &&
+            $cell_id !~ /$self->{option_results}->{filter_cell_id}/);
+
+        $self->{cells}->{$instance} = {
+            cellId => $cell_id,
+            simId  => $result->{simId},
+            status => {
+                cellId => $cell_id,
+                imsi   => $result->{imsi},
+                simId  => $result->{simId},
+                interfaceState => $result2->{interfaceState}
+            },
+            signal => { cellId => $cell_id }
+        };
+    }
+
+    if (scalar(keys %{$self->{cells}}) <= 0 && defined($self->{option_results}->{filter_cell_id}) && $self->{option_results}->{filter_cell_id} ne '') {
```

Comment:
@omercier Avoid OK: Empty output if the filter didn't catch anything (see below). But maybe in link with the overall issue with OK: Empty outputs.

```
# Return : OK:  | 'modules.cellradio.detected.count'=0;;;0;
    # return if (scalar(keys %{$self->{cells}}) <= 0);
```